build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/RenameTestCases.cs

```diff
@@ -134,28 +144,49 @@ static void CreateErrorResultsFile (string source, string dest, Exception e)
 						new XAttribute ("current-uiculture", "en-US")),
 					new XElement ("test-suite",
 						new XAttribute ("type", "APK-File"),
-						new XAttribute ("name", dest),
+						new XAttribute ("name", testSuiteName),
 						new XAttribute ("executed", "True"),
 						new XAttribute ("result", "Failure"),
 						new XAttribute ("success", "False"),
 						new XAttribute ("time", "0"),
 						new XAttribute ("asserts", "0"),
 						new XElement ("results",
 							new XElement ("test-case",
-								new XAttribute ("name", Path.GetFileName (dest)),
+								new XAttribute ("name", testCaseName),
 								new XAttribute ("executed", "True"),
 								new XAttribute ("result", "Error"),
 								new XAttribute ("success", "False"),
 								new XAttribute ("time", "0.0"),
 								new XAttribute ("asserts", "1"),
 								new XElement ("failure",
 									new XElement ("message",
-										$"Error processing `{source}`.  " +
+										$"Error processing `{sourceFile}`.  " +
 										$"Check the build log for execution errors.{Environment.NewLine}" +
 										$"File contents:{Environment.NewLine}",
-										new XCData (contents)),
+										new XCData (contents.ToString ())),
 									new XElement ("stack-trace", e.ToString ())))))));
-			doc.Save (dest);
+			doc.Save (destFile);
+		}
+
+		// Example `SourceFile`:
+		//   /Users/builder/jenkins/workspace/xamarin-android-pr-builder-release/xamarin-android/bin/TestRelease/TestResult-Mono.Android_Tests.xml
```

Comment:
I think the test result files are in different location. For example `.../xamarin-android-pr-builder-release/ws/xamarin-android/TestResult-Mono.Android_Tests-Release.xml`

Reply:
The `../xamarin-android/TestResult-Mono.Android_Tests-Release.xml` file is *created* by the `<RenameTestCases/>` Task.

For example, [`xa-build-status-v9.1.199.54_Darwin-x86_64_pr_752908f5-Release.zip`](
https://jenkins.mono-project.com/job/xamarin-android-pr-builder-release/189/Azure/processDownloadRequest/xamarin-android-pr-builder-release-189/xamarin-android/xa-build-status-v9.1.199.54_Darwin-x86_64_pr_752908f5-Release.zip) contains the file `xa-build-status-v9.1.199.54_Darwin-x86_64_pr_752908f5-Release/bin/BuildRelease/msbuild-20181213T014423-Target-RunApkTests.binlog`, which contains:

```
Task "RenameTestCases"
  Task Parameter:Configuration=Release-Aot
  Task Parameter:DeleteSourceFiles=True
  Task Parameter:DestinationFolder=/Users/builder/jenkins/workspace/xamarin-android-pr-builder-release/xamarin-android/build-tools/scripts/../..
  Task Parameter:SourceFile=/Users/builder/jenkins/workspace/xamarin-android-pr-builder-release/xamarin-android/tests/../bin/TestRelease/TestResult-Mono.Android_Tests.xml
    [Output] CreatedFiles:
      [Output] /Users/builder/jenkins/workspace/xamarin-android-pr-builder-release/xamarin-android/build-tools/scripts/../../TestResult-Mono.Android_Tests-Release-Aot.xml:
Done executing task "RenameTestCases".
```

Reply:
These are binlog files though. Here we work with test result files (starting with `TestResult`), which I don't see in the workspace of master nor PR builders.

Reply:
Ah, I misread the above. So the file ends in `xamarin-android` after the `RenameTestCases` is run, while the code in this PR runs before the move.